lib/surface/compiler/eex_engine.ex

```diff
@@ -399,16 +400,10 @@ defmodule Surface.Compiler.EExEngine do
     !!value
   end
 
-  defp to_prop_expr([%AST.AttributeExpr{value: expr}], :boolean) do
-    quote generated: true do
-      !!unquote(expr)
-    end
+  defp to_prop_expr([%AST.AttributeExpr{value: value, meta: meta}], type) do
+    Surface.TypeHandler.update_prop_expr(type, value, meta)
```

Comment:
I have the feeling we might not even need the `update_prop_expr` callback at all. What if we just use the `expr_to_quoted` callback to generate different expressions for attributes and properties based on the `module` argument? For instance:

```elixir
def expr_to_quoted(type, attribute_name, clauses, opts, module, original) do
  ast = 
    quote generated: true do
      Surface.TypeHandler.expr_to_value!(
        unquote(type),
        unquote(attribute_name),
        unquote(clauses),
        unquote(opts),
        unquote(module),
        unquote(original)
      )
      
  if module do
    # For properties
    quote generated: true do
      some_wrapper_function(unquote(ast))
    end
  else
    # For attributes
    ast
  end
end
```

I also think we might need to pass a few options to this function to enable the caller to customize the behaviour. For instance, we might have a directive that requires the attribute of type `:list` to be wrapped in `Surface.attr` because it will be treated as runtime value, however, another directive may require the original unwrapped value so it can manipulate the AST at compile-time.

Another alternative is to have separate callbacks for properties and attributes like `attr_expr_to_quoted` and `prop_expr_to_quoted`. I'm not sure yet which is one is better. We'll need to try it out to find out :)